src/lib/support/tests/TestIniEscaping.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/support/IniEscaping.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+using namespace chip;
+using namespace chip::IniEscaping;
+
+namespace {
+
+struct TestCase
+{
+    const char * input;
+    const char * expectedOutput;
+};
+
+void TestEscaping(nlTestSuite * inSuite, void * inContext)
+{
+    NL_TEST_ASSERT(inSuite, EscapeKey("") == "");
+    NL_TEST_ASSERT(inSuite, EscapeKey("abcd1234,!") == "abcd1234,!");
+    NL_TEST_ASSERT(inSuite, EscapeKey("ab\ncd =12\\34\x7f") == "ab\\x0acd\\x20\\x3d12\\x5c34\\x7f");
```

Comment:
Should the expected string be `ab\\x0acd\\x20\\x3d12\\x5c\\x5c34\\x5c7f`

Reply:
No, because the inner `\\` between 12 and 34 is 1 character